src/payment-upon-iot/lib/logic.ergo

```diff
@@ -0,0 +1,167 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace org.accordproject.payment.iot
+
+import org.accordproject.cicero.runtime.*
+import org.accordproject.iot.*
+
+import org.accordproject.signature.ContractSigned
+import org.accordproject.payment.PaymentReceived
+import org.accordproject.money.MonetaryAmount
+
+/**
+ * This smart clause emits a payment obligation when the contract is signed
+ */
+contract PaymentUponCounter over PaymentUponButtonContract state CounterState {
+  
+	/**
+	 * Sets the initial state for the clause
+	 */
+  clause init(request : Request) : Response {		
+		set state CounterState {
+			stateId: "1",
+			status: "INITIALIZED",
+			counter: 0.0,
+			paymentCount: 0.0
+		};
+
+    return CounterResponse{
+		counter: state.counter,
+		paymentCount: state.paymentCount,
+	}
+  }
+
+	/**
+	 * When the contract is signed a payment obligation is emitted
+	 */
+  clause contractSigned(request : ContractSigned) : Response emits PaymentObligation {
```

Comment:
The clause does not emit obligations, should it? Otherwise, the `emits PaymentObligation` should be removed.